include/dlaf/solver/triangular/impl.h

```diff
@@ -714,6 +714,91 @@ void Triangular<backend, device, T>::call_LUN(comm::CommunicatorGrid grid, blas:
   }
 }
 
+template <Backend backend, Device D, class T>
+void Triangular<backend, D, T>::call_LUT(comm::CommunicatorGrid grid, blas::Op op, blas::Diag diag,
+                                         T alpha, Matrix<const T, D>& mat_a, Matrix<T, D>& mat_b) {
+  using namespace triangular_lut;
+  using pika::threads::thread_priority;
+
+  auto executor_mpi = dlaf::getMPIExecutor<backend>();
+
+  common::Pipeline<comm::Communicator> mpi_row_task_chain(grid.rowCommunicator().clone());
+  common::Pipeline<comm::Communicator> mpi_col_task_chain(grid.colCommunicator().clone());
+
+  const comm::Index2D this_rank = grid.rank();
+
+  const matrix::Distribution& distr_a = mat_a.distribution();
+  const matrix::Distribution& distr_b = mat_b.distribution();
+
+  if (mat_b.size().isEmpty())
+    return;
+
+  constexpr std::size_t n_workspaces = 2;
+  common::RoundRobin<matrix::Panel<Coord::Col, T, D>> a_panels(n_workspaces, distr_a);
+  common::RoundRobin<matrix::Panel<Coord::Row, T, D>> b_panels(n_workspaces, distr_b);
+
+  for (SizeType k = 0; k < mat_a.nrTiles().rows(); ++k) {
+    const GlobalTileIndex kk{k, k};
+
+    const LocalTileIndex kk_offset{distr_a.nextLocalTileFromGlobalTile<Coord::Row>(kk.row() + 1),
+                                   distr_a.nextLocalTileFromGlobalTile<Coord::Col>(kk.col())};
+    const LocalTileIndex bt_offset(distr_b.nextLocalTileFromGlobalTile<Coord::Row>(kk.row()), 0);
+
+    auto& a_panel = a_panels.nextResource();
+    auto& b_panel = b_panels.nextResource();
+
+    if (kk.row() == mat_a.nrTiles().rows() - 1) {
+      a_panel.setWidth(mat_a.tileSize(kk).rows());
+      b_panel.setHeight(mat_a.tileSize(kk).cols());
+    }
+
+    const auto rank_kk = distr_a.rankGlobalTile(kk);
+    if (this_rank.col() == rank_kk.col()) {
+      for (SizeType i_loc = kk_offset.row() - 1; i_loc >= 0; --i_loc) {
+        const LocalTileIndex ik{i_loc, kk_offset.col()};
+        a_panel.setTile(ik, mat_a.read(ik));
+      }
+    }
+    comm::broadcast(executor_mpi, rank_kk.col(), a_panel, mpi_row_task_chain);
+
+    matrix::util::set0<backend>(thread_priority::normal, b_panel);
+
+    for (const auto& ij :
+         common::iterate_range2d(LocalTileIndex{0, 0},
+                                 LocalTileIndex{bt_offset.row(), distr_b.localNrTiles().cols()}))
```

Comment:
Since the start index is hard-coded to be the origin, you can use the helper `iterate_range2d` that accepts just the end index.